src/zcbenchmarks.cpp

```diff
@@ -420,3 +453,15 @@ double benchmark_sendtoaddress(CAmount amount)
     return timer_stop(tv_start);
 }
 
+double benchmark_loadwallet()
+{
+    pre_wallet_load();
```

Comment:
This approach is good enough for now. However, I'm somewhat worried that loading the full machinery of a node, then trying to shutdown portions of it, all in order to isolate the `LoadWallet` call below, is very noisy for a benchmark. To me this seems like we're shoe-horning the logic to fit other benchmarks and the risk is various kinds of caching or hard-to-control global state interferes with a good measurement. Hopefully if that is truly a problem we can notice it in our measurements and improve things later.

Reply:
The way our benchmarks are currently designed, each "run" of a benchmark happens inside the same (regtest) `zcashd` process, but separate benchmarks happen in separate clean instances. So this shutdown/restart logic is only there for two reasons:

- Insulate independent wallet loading benchmark runs from each other (as much as possible).
- Defend against unexpected uses of this code (e.g. someone downstream removing the regtest requirement and trying it on their live wallet).

I agree it would be nice to find a cleaner way to do this, that still enables us to measure exactly what we need. We should see how well the benchmark reflects reality once merged.

Reply:
Another way we could do this is to just start the node, and extract the wallet loading time from the `debug.log`. This would be conceptually simpler, and even closer to the measuerments we are after. However, it would probably be slower overall (as it would require cycling the entire `zcashd` node instead of just the wallet part), and would be so different to our current benchmarks that it would need to use its own external setup and control mechanism instead of `performance-measurements.sh`.

Reply:
(We'd also need to improve the accuracy of the `debug.log` number, probably by turning our benchmark timing functions into utilities.)

Reply:
I don't really understand the motivation for doing it this way, but I guess it's ok.

Reply:
Key motivation is that if you spin up a new wallet inside the node, you get incorrect (doubled) memory readings, as well as likely-inconsistent time readings due to the load of two wallets in-process. That said, I don't know whether there might be other unaccounted-for effects of doing it this way, but if there are, we can address them once we start observing them in the benchmarks.